src/lib/battery/battery.cpp

```diff
@@ -268,25 +271,25 @@ void Battery::computeScale()
 
 float Battery::computeRemainingTime(float current_a)
 {
-	float time_remaining_s{NAN};
-
-	// Only estimate remaining time with useful in flight current measurements
-	if (_current_filter_a.getState() > 1.f) {
-		// Initialize strongly filtered current to an estimated average consumption
-		if (_current_average_filter_a.getState() < 0.f) {
-			// TODO: better initial value based on "average current" from last flight
-			_current_average_filter_a.reset(15.f);
-		}
+	float time_remaining_s = NAN;
 
-		// Filter current very strong, we basically want the average consumption
-		_current_average_filter_a.update(current_a);
+	// Remaining time estimation only possible with capacity
+	if (_params.capacity > 0.f) {
+		vehicle_status_s vehicle_status{};
+		_vehicle_status_sub.copy(&vehicle_status);
 
-		// Remaining time estimation only possible with capacity
-		if (_params.capacity > 0.f) {
-			const float remaining_capacity_mah = _state_of_charge * _params.capacity;
-			const float current_ma = _current_average_filter_a.getState() * 1e3f;
-			time_remaining_s = remaining_capacity_mah / current_ma * 3600.f;
+		if (!PX4_ISFINITE(_current_average_filter_a.getState()) || _current_average_filter_a.getState() < FLT_EPSILON) {
```

Comment:
Couldn't the filter in principle end up having a negative average current?
I think we should define unknown current values as NAN instead of a negative value. But let's not do that in this pr.

Reply:
To be solved in another pr.